### PR TITLE
Sleeping changes

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:

Citizens now go to sleep earlier/later depending on distance to go. They also do complain with an interaction if their way is too long(work-home).
Research effects on entities are now applied upon initializing the entity, or upon finishing research rather than each X ticks.
Overhaul sleeping AI, fixing some possible bugs
Overhaul Citizenentity desiredTask and livingticks
Add simplenotification interaction, which disappears after clicking ignore or "I will work on it".

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenDiseaseHandler.java [hunk 1]
```diff
@@ -56,7 +55,7 @@ public CitizenDiseaseHandler(final AbstractEntityCitizen citizen)
     @Override
     public void tick()
     {
-        if (citizen.getTicksExisted() % TICKS_20 == 0 && citizen.getCitizenColonyHandler().getColony().isActive() && !(citizen.getCitizenJobHandler()
+        if (citizen.getCitizenColonyHandler().getColony().isActive() && !(citizen.getCitizenJobHandler()
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenDiseaseHandler.java:58]: Do we want to tick this so often?

Comment (someaddons) [reply src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenDiseaseHandler.java:58]: still ticks every 20, just called every 20 now

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/CitizenDiseaseHandler.java:58]: ah ok

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/ldtteam/minecolonies/pull-request/5593/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ldtteam/minecolonies)

**Review by marchermans — APPROVED** [review]

**Pull request merged**